components/prism-purescript.js

```diff
@@ -0,0 +1,38 @@
+Prism.languages.purescript = {
+	'comment': {
+		pattern: /(^|[^-!#$%*+=?&@|~.:<>^\\\/])(?:--[^-!#$%*+=?&@|~.:<>^\\\/].*|{-[\s\S]*?-})/m,
+		lookbehind: true
+	},
+	'char': /'(?:[^\\']|\\(?:[abfnrtv\\"'&]|\^[A-Z@[\]^_]|NUL|SOH|STX|ETX|EOT|ENQ|ACK|BEL|BS|HT|LF|VT|FF|CR|SO|SI|DLE|DC1|DC2|DC3|DC4|NAK|SYN|ETB|CAN|EM|SUB|ESC|FS|GS|RS|US|SP|DEL|\d+|o[0-7]+|x[0-9a-fA-F]+))'/,
+	'string': {
+		pattern: /"(?:[^\\"]|\\(?:[abfnrtv\\"'&]|\^[A-Z@[\]^_]|NUL|SOH|STX|ETX|EOT|ENQ|ACK|BEL|BS|HT|LF|VT|FF|CR|SO|SI|DLE|DC1|DC2|DC3|DC4|NAK|SYN|ETB|CAN|EM|SUB|ESC|FS|GS|RS|US|SP|DEL|\d+|o[0-7]+|x[0-9a-fA-F]+)|\\\s+\\)*"/,
```

Comment:
This pattern has an exponential runtime for certain input strings for two reasons:

1. The inner loop can match `\SOH` in two ways: 1) it can be matched whole (via the `SOH` alternative) and 2) it can be matched as `\SO` (via the `SO` alternative) followed by an `H` character (using `[^\\"]`). This means that a string of the form `/"(?:\\SOH)+/` will cause exponential backtracking. ([Try it out](https://regexr.com/5avbu).)
2. The `\d+|o[0-7]+|x[0-9a-fA-F]+` will all cause exponential backtracking for a similar reason.

I think the easiest fix here, is to not validate escape sequences.

Note: The `char` pattern is safe because it doesn't repeat characters.

```suggestion
		pattern: /"(?:[^\\"]|\\(?\S|\s+\\)*"/,
```